pytest_html/plugin.py

```diff
@@ -105,12 +105,51 @@ def __init__(self, outcome, report, logfile, config):
             self.logfile = logfile
             self.config = config
             self.row_table = self.row_extra = None
+            self.images = []
 
             test_index = hasattr(report, 'rerun') and report.rerun + 1 or 0
 
             for extra_index, extra in enumerate(getattr(report, 'extra', [])):
                 self.append_extra_html(extra, extra_index, test_index)
 
+            num_of_images = len(self.images)
+            if num_of_images == 1:  # Only one image, no need for carousel
```

Comment:
Have you considered generalizing it using one for loop instead of the if elif else stuff.